src/displayapp/screens/settings/SettingSetTime.cpp

```diff
@@ -105,38 +121,90 @@ SettingSetTime::SettingSetTime(Pinetime::Applications::DisplayApp *app, Pinetime
   lv_obj_align(btnSetTime, lv_scr_act(), LV_ALIGN_IN_BOTTOM_MID, 0, 0);
   lv_obj_set_style_local_value_str(btnSetTime, LV_BTN_PART_MAIN, LV_STATE_DEFAULT, "Set");
   lv_obj_set_event_cb(btnSetTime, event_handler);
+
+  if (settingsController.GetClockType() == Controllers::Settings::ClockType::H12) {
+    struct Time12H convertedTime = timeConvert(hoursValue);
+    if (hoursValue < 12) {
+      lv_label_set_text_static(lblampm, "AM");
+    } else {
+      lv_label_set_text_static(lblampm, "PM");
+    }
+    lv_label_set_text_fmt(lblHours, "%02d", convertedTime.hours);
+  }
 }
 
 SettingSetTime::~SettingSetTime() {
   lv_obj_clean(lv_scr_act());
 }
 
-void SettingSetTime::HandleButtonPress(lv_obj_t *object, lv_event_t event) {
+Time12H timeConvert(int time24H) {
+  struct Time12H time12H;
+  switch (time24H) {
+    case 0:
+      time12H.hours = 12;
+      time12H.ampm = "AM";
+      break;
+    case 1 ... 11:
+      time12H.hours = time24H;
+      time12H.ampm = "AM";
+      break;
+    case 12:
+      time12H.hours = 12;
+      time12H.ampm = "PM";
+      break;
+    case 13 ... 23:
+      time12H.hours = time24H - 12;
+      time12H.ampm = "PM";
+      break;
+  }
+  return time12H;
+}
+
+void SettingSetTime::HandleButtonPress(lv_obj_t* object, lv_event_t event) {
+  bool is24H;
   if (event != LV_EVENT_CLICKED)
     return;
 
+  is24H = (settingsController.GetClockType() == Controllers::Settings::ClockType::H24);
+
   if (object == btnHoursPlus) {
     hoursValue++;
-    if (hoursValue > 23)
+    if (hoursValue > 23) {
       hoursValue = 0;
-    lv_label_set_text_fmt(lblHours, "%02d", hoursValue);
+    }
+    struct Time12H convertedTime = timeConvert(hoursValue);
+    if (!is24H) {
+      lv_label_set_text_static(lblampm, convertedTime.ampm.c_str());
+      lv_label_set_text_fmt(lblHours, "%02d", convertedTime.hours);
+    } else {
+      lv_label_set_text_fmt(lblHours, "%02d", hoursValue);
+    }
```

Comment:
This block of code is still duplicated above and below. +125..+133 does pretty much the same thing as this block of code.

Reply:
I had originally replaced lines 125-133 with`lv_label_set_text_static(lblampm, convertedTime.ampm.c_str());`, however with this, the ampm label was not populated until an hour button was pressed. The only resolution I found was to use an if statement instead. I will continue working on a better fix.

Reply:
Issue is resolved. The if else is replaced by `lv_label_set_text_static(lblampm, convertedTime.ampm)` and will be included in the next commit.

Reply:
I meant to suggest moving the LVGL functions into the function as well. This way you don't need to return anything and the blocks of code can be replaced by a single function. The function would be called something like `SetHourLabels()`.

Reply:
That is a much more elegant solution. I will implement that for the next commit.